plugins/by-name/lspsaga/deprecations.nix

```diff
@@ -0,0 +1,205 @@
+lib: {
+  imports =
+    let
+      basePluginPath = [
+        "plugins"
+        "lspsaga"
+      ];
+    in
+    map
+      (
+        optionName:
+        # See https://github.com/NixOS/nixpkgs/pull/247916
+        # TODO remove those warnings in ~2 months (October 2023)
+        lib.mkRemovedOptionModule (basePluginPath ++ [ optionName ]) ''
+          The `lspsaga` plugin used by default in Nixvim has changed to https://github.com/nvimdev/lspsaga.nvim.
+          Please, adapt your config to the new options.
+          Documentation available at https://nix-community.github.io/nixvim/plugins/lspsaga/index.html.
+        ''
+      )
+      [
+        "signs"
+        "headers"
+        "maxDialogWidth"
+        "icons"
+        "maxFinderPreviewLines"
+        "keys"
+        "borderStyle"
+        "renamePromptPrefix"
+      ]
+    ++ [
+      # TODO: introduced 2025-08-20: remove after 25.11
+      (
+        let
+          oldOptPath = basePluginPath ++ [
+            "finder"
+            "filter"
+          ];
+        in
+        lib.mkChangedOptionModule oldOptPath
+          (
+            basePluginPath
+            ++ [
+              "settings"
+              "finder"
+              "filter"
+            ]
+          )
+          (
+            config:
+            let
+              oldFilter = lib.getAttrFromPath oldOptPath config;
+            in
+            lib.nixvim.ifNonNull' oldFilter (lib.mapAttrs (_: lib.nixvim.mkRaw) oldFilter)
+          )
+      )
+
+      # TODO: added 2024-09-20 remove after 24.11
+      (
+        { config, ... }:
+        let
+          cfg = config.plugins.lspsaga;
+        in
+        {
+          plugins.web-devicons =
+            lib.mkIf
+              (
+                (cfg.settings.ui.devicon or true)
+                && !(
+                  config.plugins.mini.enable
+                  && config.plugins.mini.modules ? icons
+                  && config.plugins.mini.mockDevIcons
+                )
+              )
+              {
+                enable = lib.mkOverride 1490 true;
+              };
+        }
+      )
```

Comment:
I believe it was a mistake that this isn't conditional on `cfg.enable`.

If so, this could be moved to `extraConfig` (if preferred), or `cfg.enable &&` could be added here.

Alternatively, we've probably had this in enough release versions that it could be dropped now (although our deprecation lifespan policy discussion is still unresolved)

Reply:
Actually, it must've been conditional before as this is now causing CI to fail. ([e.g.](https://buildbot.nix-community.org/#/builders/2908/builds/5301))